From 8936ccef4851a41151309dc51d47c64d8215b561 Mon Sep 17 00:00:00 2001
From: alopez <angellopezcima@aurigae.com>
Date: Wed, 26 Nov 2014 14:58:02 +0100
Subject: [PATCH] Replaced the class DefaultHttpClient by
 SystemDefaultHttpClient to set the proxy using the system properties as
 described in
 https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html.

---
 src/main/java/jenkins/plugins/http_request/HttpRequest.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main/java/jenkins/plugins/http_request/HttpRequest.java b/src/main/java/jenkins/plugins/http_request/HttpRequest.java
index e3abd45..d6d86ca 100644
--- a/src/main/java/jenkins/plugins/http_request/HttpRequest.java
+++ b/src/main/java/jenkins/plugins/http_request/HttpRequest.java
@@ -21,7 +21,7 @@ import jenkins.plugins.http_request.util.RequestAction;
 import net.sf.json.JSONObject;
 import org.apache.http.HttpResponse;
 import org.apache.http.client.methods.HttpRequestBase;
-import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.impl.client.SystemDefaultHttpClient;
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
@@ -96,7 +96,7 @@ public class HttpRequest extends Builder {
         final HttpMode mode = httpMode != null ? httpMode : getDescriptor().getDefaultHttpMode();
         logger.println("HttpMode: " + mode);
 
-        final DefaultHttpClient httpclient = new DefaultHttpClient();
+        final SystemDefaultHttpClient httpclient = new SystemDefaultHttpClient();
 
         logger.println("Parameters: ");
         final EnvVars envVars = build.getEnvironment(listener);
-- 
1.9.0.msysgit.0

